### PR TITLE
Disallow globals in protocol 6.

### LINKED ISSUE #331: Our Wasm validator does not adhere to recent changes in Wasm specification.
Specifically in https://github.com/WebAssembly/spec/pull/1525 they restricted the appearance of globals in data and element sections (their initializers).

In the standard that we implemented https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#modules%E2%91%A3 this was allowed, and we adhere to the specification as stated there.

The discussion in https://github.com/WebAssembly/spec/issues/1522 indicates that they consider this a typo in the specification.

We need to figure out what we should do. Consider this a bug or keep it.

If we were to fix it this requires a protocol update since it changes criteria for validity of smart contracts.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Closes #331 

## Changes

- Introduce a configuration for module validation in wasm-transform. We will use this in a followup PR to allow/disallow new instructions.
- Propagate the changes to wasm-chain-integration. In V0 modules we keep the existing validation and behaviour.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by Bargsteen — APPROVED** [review]
LGTM 👍 Nice work

File: smart-contracts/wasm-chain-integration/src/v0/ffi.rs [hunk 1]
```diff
@@ -188,6 +191,9 @@ unsafe extern "C" fn validate_and_process_v0(
 ) -> *mut u8 {
     let wasm_bytes = slice_from_c_bytes!(wasm_bytes_ptr, wasm_bytes_len);
     match utils::instantiate_with_metering::<ProcessedImports, _>(
+        ValidationConfig {
+            allow_globals_in_init: true,
+        },
```
[the remark this answers is not in the document]
Comment (abizjak) [reply smart-contracts/wasm-chain-integration/src/v0/ffi.rs:196]: There was, in the sense that I wrote it before writing the constants.

[the remark this answers is not in the document]
Comment (abizjak) [reply smart-contracts/wasm-chain-integration/src/v0/ffi.rs:196]: :)

File: smart-contracts/wasm-transform/src/validate.rs [hunk 2]
```diff
@@ -858,9 +858,32 @@ pub trait ValidateImportExport {
     fn validate_export_function(&self, item_name: &Name, ty: &FunctionType) -> bool;
 }
 
+/// Configuration for module validation.
+/// This determines the features of Wasm we support.
```
[the remark this answers is not in the document]
Comment (abizjak) [reply smart-contracts/wasm-transform/src/validate.rs:862]: Added a bit

**Review by limemloh — APPROVED** [review]

**Pull request merged**